server/plugin.go

```diff
@@ -145,6 +145,8 @@ func (p *Plugin) GetURL() string {
 
 func (p *Plugin) OnDisconnectedTokenHandler(userID string) {
 	p.API.LogDebug("OnDisconnectedTokenHandler", "userID", userID)
+	p.metricsService.IncrementInvalidTokenDetected()
```

Comment:
To confirm, because we set the token to `nil` via `SetUserInfo` below, we won't continuously "spam count" here? Is there a chance we end up double or triple counting in some cases, skewing the metric? Not necessary for this PR, but I wonder if we should only increment the metric if `SetUserInfo` actually changes something (using the database as a mutex for counting once).

Reply:
That's a good point, I don't think it is going to happen that often, but yes, we can find the way to make this run only once.